src/builder.rs

```diff
@@ -1043,3 +1059,68 @@ fn seed_bytes_from_config(
 		},
 	}
 }
+
+/// Sanitize the user-provided node alias to ensure that it is a valid protocol-specified UTF-8 string.
+fn sanitize_alias<T: Into<String>>(node_alias: T) -> Result<String, BuildError> {
+	// Alias is convertible into UTF-8 encoded string
+	let node_alias: String = node_alias.into();
+	let alias = node_alias.trim();
+
+	// Alias is non-empty
+	if alias.is_empty() {
+		return Err(BuildError::InvalidNodeAlias("Node alias cannot be empty.".to_string()));
+	}
+
+	// Alias valid up to first null byte
+	let first_null = alias.as_bytes().iter().position(|b| *b == 0).unwrap_or(alias.len());
```

Comment:
I took a cursory look at rust-lightning and borrowed [part of their validation approach](https://github.com/lightningdevkit/rust-lightning/blob/3d76753a10a0a44a5598164b251c3ce105fb45bb/lightning/src/routing/gossip.rs#L1399) before displaying `NodeAlias`es. I thought it would be a useful validation at this level rather than allowing aliases with sandwiched `null` bytes to get to the protocol level.

Reply:
This has been removed and is not relevant.